portfolio/src/main/java/com/google/sps/servlets/Comment.java

```diff
@@ -109,33 +118,45 @@ public Entity createEntity() {
     commentEntity.setProperty("lastName", this.lastName);
     commentEntity.setProperty("email", this.email);
     commentEntity.setProperty("comment", this.comment);
+    commentEntity.setProperty("timestamp", this.timestamp);
     commentEntity.setProperty("visitReason", this.visitReason.name()); // set to enum property name
 
+    // Return the created entity
     return commentEntity;
   }
 
   /**
    * Static method to convert a datastore to an ArrayList of Comment objects.
    * @param datastore instance of DatastoreService that contains "Comment" entities
+   * @param max maximum number of comments to return. Set to 0 for no maximum
    * @return an ArrayList of comment objects with data from each entity in datastore.
    */
-  public static ArrayList<Comment> datastoreToArrayList(DatastoreService datastore) {
+  public static ArrayList<Comment> datastoreToArrayList(DatastoreService datastore, int max) {
     // Create a query
-    Query query = new Query("Comment");
+    Query query = new Query("Comment")
+        .addSort("timestamp", Query.SortDirection.DESCENDING);
 
     // Get the results from the query
-    PreparedQuery results = datastore.prepare(query);
+    PreparedQuery resultsQuery = datastore.prepare(query);
 
     // Create ArrayList to return Comment objects
     ArrayList<Comment> comments = new ArrayList<>();
 
+    // Get results as an iterable
+    Iterable<Entity> results = resultsQuery.asIterable();
+
     // Iterate through the entities
     // Create a Comment instance for each entity and add it to the ArrayList
-    for (Entity entity: results.asIterable()) {
+    // until the maximum amount of elements are created (if a maximum exists)
+    for (Entity entity: results) {
+      if(max != 0 && comments.size() >= max) {
```

Comment:
Nit: Space before parentheses if you decide to keep the loop

Reply:
Hmmm - I don't know if I see what you're describing. As of now there currently is a space between for and (Entity...)

Also, I think I have to keep the loop as I still need to cast the Entity objects into Comment objects no matter what. I could replace with a forEach, but that's basically the same thing

Reply:
Space between `if` and `(max...`.

You could do it in a more functional programming fashion via `stream().map()` (https://mkyong.com/java8/java-8-streams-map-examples/), which I think often makes things clearer, but it ends up being a matter of style. So feel free to decide if that makes sense here or not.